### DISCUSSION TITLE
Compile on stm32F207

### DISCUSSION THREAD

**Discussion by olekdata** [bigtreetech/BIGTREETECH-TouchScreenFirmware#1600] [Q&A]
Can TFT 3.5 be uploaded soft "normally" using SWD (e.g. ST-LINK V2) and VSC + PlatformIO? Do you have to upload it via SD card?

The program compiles with the settings:

  _[env:BIGTREE_TFT35_V3_0]
  platform      = ststm32@9.0.0
  framework     = stm32cube
  board         = STM32F103VC
   upload_protocol = cmsis-dap_

but with the stm32F207, not anymore

  _[env:BIGTREE_TFT35_V3_0]
  platform      = ststm32@9.0.0
  framework     = stm32cube
  board         = STM32F207VC
  upload_protocol = cmsis-dap_

I have TFT 3.5 V3.0 with STM32F207. Can I upload firmware version compiled on STM32F103 ? Probably not.

**Comment by bigtreetech** [comment]
https://github.com/bigtreetech/BIGTREETECH-TouchScreenFirmware/pull/1562
This PR will fix it. but the ".ld" files should be separated for every borad.
Still working...
And when you use the SWD interface to upload, please be careful not to erase the bootloader

**Reply by olekdata** [reply]
I probably have a different problem when compiling

a lot of conflicts : e.g.
  _...stm32f2xx_hal_dcmi.h:71:2: error: conflicting types for 'DCMI_CodesInitTypeDef'
  ...stm32f2xx_hal_dcmi.h:101:2: error: conflicting types for 'DCMI_InitTypeDef'
  ...stm32f2xx_dac.h:93:0: note: this is the location of the previous definition  #define IS_DAC_TRIGGER(TRIGGER) (((TRIGGER) == DAC_Trigger_None) || \
  ...stm32f2xx_dcmi.h:103:3: note: previous declaration of 'DCMI_CodesInitTypeDef' was here } DCMI_CodesInitTypeDef;_

I think I have uploaded something wrong, the device does not get up - dark screen :-(

**Reply by olekdata** [reply]
Once again me.

Short questions on how to fix the display

- what to upload.
- and how to upload.
 

I can see F207 via SWD and STM32 ST-LINK Utility
![image](https://user-images.githubusercontent.com/54954537/106887234-289f3b00-66e5-11eb-8bfc-fae64f69b35e.png)

I will wait for other matters
